ui/src/components/BeamlineCamera/BeamlineCamera.jsx

```diff
@@ -1,127 +1,129 @@
+        ) : null,
       );
     }
+    return DraggableElements;
+  }
 
-    return (
-      <OverlayTrigger
-        show={showLabelOverlay}
-        rootClose
-        trigger="click"
-        placement="bottom"
-        overlay={optionsOverlay}
-      >
-        <div onClick={() => setShowLabelOverlay(!showLabelOverlay)}>
-          <Badge
-            bg="secondary"
-            style={{ display: 'block', marginBottom: '3px' }}
-          >
-            {labelText}
-            <i className="fas fa-cog ms-2" />
-          </Badge>
-        </div>
-      </OverlayTrigger>
-    );
-  };
-
-  const msgLabelStyle = {
-    display: 'block',
-    fontSize: '100%',
-    borderRadius: '0px',
-    color: '#000',
-  };
+  function renderCamera() {
+    const dropdownItem = [];
+    for (const [key, camera] of cameraSetup.components.entries()) {
+      dropdownItem.push(
+        <Dropdown.Item
+          key={key}
+          onClick={() => handleShowVideoModal(key, true)}
+        >
+          {camera.label} <i className="fas fa-video" />
+        </Dropdown.Item>,
+        <Dropdown.Divider />,
+      );
+    }
+    return dropdownItem;
+  }
 
-  const video = (
-    <div>
-      {format !== 'mp4' ? (
-        <img // eslint-disable-line jsx-a11y/no-noninteractive-element-interactions
-          onClick={onImageClick}
-          src={url}
-          alt={labelText}
-          width={width}
-          height={height}
-        />
-      ) : (
-        <video
-          src={url}
-          alt={labelText}
-          onClick={onImageClick}
-          width={width}
-          height={height}
-        />
-      )}
-      <Button
+  return cameraSetup?.components.length > 0 ? (
+    <>
+      <Dropdown
+        title="Beamline Cameras"
+        id="beamline-cameras-dropdown"
         variant="outline-secondary"
-        onClick={onImageClick}
-        size="sm"
-        style={{ position: 'absolute', left: '90%', bottom: '10px' }}
-      >
-        <img src={pip} alt="PIP Icon" />
-      </Button>
-    </div>
-  );
-
-  return (
-    <div className="inout-switch">
-      {renderLabel()}
-      <OverlayTrigger
-        show={showValueOverlay}
-        rootClose
-        trigger="click"
-        placement="bottom"
-        overlay={
-          <Popover style={{ padding: '0.5em' }} id={`${labelText} popover`}>
-            {video}
-          </Popover>
-        }
+        autoClose="outside"
+        key="beamline-cameras-dropdown"
       >
-        <div onClick={() => setShowValueOverlay(!showValueOverlay)}>
-          <Badge bg="success" style={msgLabelStyle}>
-            <i className="fas fa-video" />
-          </Badge>
-        </div>
-      </OverlayTrigger>
-    </div>
-  );
+        <Dropdown.Toggle
+          variant="outline-secondary"
+          size="sm"
+          className="mb-1"
+          style={{ width: '150px' }}
+        >
+          Beamline Cameras
+        </Dropdown.Toggle>
+        <Dropdown.Menu>{renderCamera()}</Dropdown.Menu>
```

Comment:
Embrace JSX, don't be scared! :stuck_out_tongue_closed_eyes: `render*` function should be avoided if possible; either inline it or turn it into a proper component (that's what React is all about :wink:).

Here, for the drop-down options, I think inlining is perfectly fine (see suggestion below); but I reiterate that the draggable video modal is worth turning into a component (with the loop that renders them inlined below) — sorry if you're already working on this and I'm jumping the gun.

```suggestion
        <Dropdown.Menu>{
          cameraSetup.components.entries().map(([key, camera]) => (
            <>
              <Dropdown.Item
                key={key}
                onClick={() => handleShowVideoModal(key, true)}
              >
                {camera.label} <i className="fas fa-video" />
              </Dropdown.Item>
              <Dropdown.Divider />
            </>
          );
        }</Dropdown.Menu>
```